Lib/test/_test_multiprocessing.py

```diff
@@ -5536,6 +5536,42 @@ def test_too_long_name_resource(self):
         with self.assertRaises(ValueError):
             resource_tracker.register(too_long_name_resource, rtype)
 
+    def _test_resource_tracker_leak_resources(self, context, delete_queue):
+        from multiprocessing.resource_tracker import _resource_tracker
+        _resource_tracker.ensure_running()
+        self.assertTrue(_resource_tracker._check_alive())
+
+        # Reset exit code value
+        _resource_tracker._exitcode = None
+
+        mp_context = multiprocessing.get_context(context)
+
+        # Keep it on variable, so it won't be cleared yet
+        q = mp_context.Queue()
+        if delete_queue:
+            del q
```

Comment:
Do we want to add a `gc.collect` call for non-refcounted implementations?
Or perhaps, instead of relying on Python GC semantics, perhaps explicitly close the queue?
@iritkatriel What do you think?

Reply:
We could do both.

Reply:
done